src/cmd_handlers.c

```diff
@@ -1011,6 +1016,44 @@ goto_cmd(const cmd_info_t *cmd_info)
 	return 0;
 }
 
+static int
+regedit_cmd(const cmd_info_t *cmd_info)
+{
+	if (cmd_info->argc > 1)
+		return CMDS_ERR_TRAILING_CHARS;
+
+	const int regname = cmd_info->argc > 0 ? cmd_info->argv[0][0] : DEFAULT_REG_NAME;
+	if (regname == BLACKHOLE_REG_NAME)
+	{
+		ui_sb_err("Cannot modify blackhole register.");
+		return 1;
+	}
+
+	reg_t *reg = regs_find(tolower(regname));
+	if (NULL == reg || reg->nfiles == 0)
+	{
+		ui_sb_err("Nothing to modify in the empty register.");
+		return 1;
+	}
+	const int fname_len = PATH_MAX + 1;
+	char *tmp_fname = malloc((fname_len) * sizeof(char));
```

Comment:
It's fine to have buffer on the stack.  In case you've changed implementation to make `vim_edit_files()` work, just use `vim_view_file()`.